Recruit-Training/Advanced-Recruit-Training/Embedded-Teaser/Training/Test/TrainingTest.c

```diff
@@ -5,24 +5,56 @@
 void runTrainingTests()
 {
     // Run all the tests here using RUN_TEST()
+    RUN_TEST(test_EverythingValid);
+    RUN_TEST(test_EverythingInvalid);
+    RUN_TEST(test_OnlyLightsInvalid);
+    RUN_TEST(test_OnlyMotorsInvalid);
 }
 
 void test_EverythingValid()
 {
-
+	uint8_t allValidData[] = {  0b10111111,		// Motor on, going forward with positive velocity
+								0b10111111,		// Motor on, going forward with positive velocity
+								0b01111100 };	// High headlights, with hazards on, brakes off
+	trainingTask(allValidData);
+	TEST_ASSERT_EQUAL_INT8_MESSAGE(validData, 7, "Function did not set validData to 0b111 for all valid");				// Expecting 7 for all valid
+	TEST_ASSERT_EQUAL_INT8_MESSAGE(outputArray[0], 1, "Function did not set outputArray[0] to 1 for valid motor 1");	// Expecting 1 for motor 1
+	TEST_ASSERT_EQUAL_INT8_MESSAGE(outputArray[1], 1, "Function did not set outputArray[1] to 1 for valid motor 2");	// Expecting 1 for motor 2
+	TEST_ASSERT_EQUAL_INT8_MESSAGE(outputArray[2], 1, "Function did not set outputArray[2] to 1 for valid lights");		// Expecting 1 for lights
```

Comment:
I think you might've misunderstood what the `outputArray` is suppose to contain. If the input datas are valid then they are copied into the `outputArray`.  So in this case since everything is valid, `outputArray[0]` should equal `0b10111111`,  `outputArray[1]` should equal `0b10111111` and `outputArray[2]` should equal `0b01111100`.